modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/request_controller.rb

```diff
@@ -33,28 +33,28 @@ def index
 
         def decide
           proc_id = form_attributes['procId']
+          ptcpnt_id = form_attributes['ptcpntId']
+          decision = normalize(form_attributes['decision'])
 
-          unless proc_id
-            raise ::Common::Exceptions::ParameterMissing.new('procId',
-                                                             detail: 'procId is required')
-          end
-
-          decision = form_attributes['decision']
-
-          unless decision && %w[accepted declined].include?(normalize(decision))
-            raise ::Common::Exceptions::ParameterMissing.new(
-              'decision',
-              detail: 'decision is required and must be either "ACCEPTED" or "DECLINED"'
-            )
-          end
+          validate_decide_params!(proc_id:, decision:)
 
           service = ManageRepresentativeService.new(external_uid: 'power_of_attorney_request_uid',
                                                     external_key: 'power_of_attorney_request_key')
 
+          poa_request = validate_ptcpnt_id!(ptcpnt_id:, proc_id:, service:) if decision == 'declined'
+
+          if poa_request
+            first_name = poa_request['claimantFirstName'] || poa_request['vetFirstName'].presence ||
+                         poa_request['VSOUserFirstName'].presence
+            poa_code = poa_request['poaCode']
+          end
+
           res = service.update_poa_request(proc_id:, secondary_status: decision,
                                            declined_reason: form_attributes['declinedReason'])
 
-          raise ::Common::Exceptions::Lighthouse::BadGateway unless res
+          raise Common::Exceptions::Lighthouse::BadGateway if res.blank?
+
+          send_declined_notification(ptcpnt_id:, first_name:, poa_code:) if decision == 'declined'
```

Comment:
should we just pass the ptcpnt_id here and make a similar call to the controller internal the job? just for less PII? doesn't really save us much, but we could get the details from BGS in the job and have access to first_name and poa_code from that. Really wish we had a way to call BGS with just the procId.....

Reply:
Since we still have to send encrypted PII (participant ID) and validate the request in the controller anyway, I figure there is no need to duplicate BGS calls just to save a param or two.